src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewModel_GetModelUpdates.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Collections;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    internal partial class DocumentOutlineViewModel
+    {
+        /// <summary>
+        /// Queue that uses the language-server-protocol to get document symbol information.
+        /// This queue can return null if it is called before and LSP server is registered for our document.
+        /// </summary>
+        private readonly AsyncBatchingResultQueue<DocumentSymbolDataModel> _documentSymbolQueue;
+
+        private async ValueTask<DocumentSymbolDataModel> GetDocumentSymbolAsync(CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var textBuffer = _textBuffer;
+            var filePath = _textBuffer.GetRelatedDocuments().FirstOrDefault(static d => d.FilePath is not null)?.FilePath;
```

Comment:
can this just be obtained in the constructor, and not here?

Reply:
File path can be changed at any time we cannot cache it.